### ISSUE TITLE
CacheUnmanagedCertificatePEMFile rejects duplicate certs, without accounting for tags

### ISSUE THREAD

**Issue by Bun** [caddyserver/certmagic#211]
## What version of the package are you using?
 v0.17.2

## What are you trying to do?
I originally ran into this issue when porting an existing nginx setup to Caddy:

I have a pre-existing LE certificate, with multiple SANs names. This certificate is matched based on tags. When CacheUnmanagedCertificatePEMFile encounters a certificate it has already loaded, it rejects it ("certificate already cached"), which ignores any "additional" tags assigned to the certificate.

## What steps did you take?

In Caddy, I create a config similar to:
```
sub1.domain {
    tls /var/lib/acme/live/sub1.domain/fullchain /var/lib/acme/live/sub1.domain/privkey
}
sub2.domain {
    tls /var/lib/acme/live/sub2.domain/fullchain /var/lib/acme/live/sub2.domain/privkey
}
```
Note that the filepaths aren't the same, even though the certificates actually are identical. (The subdomain directories actually point to the same on-disk directory, since both subdomains are included in the certificate.)

From this, Caddy generates something like:
```
"tls": {
    "certificates": {
        "load_files": [
            {"certificate": "/var/lib/acme/live/sub1.domain/fullchain", "key": "/var/lib/acme/live/sub1.domain/privkey", "tags": ["cert1"]},
            { "certificate": "/var/lib/acme/live/sub2.domain/fullchain", "key": "/var/lib/acme/live/sub2.domain/privkey","tags": ["cert0"]}
        ]
    }
}
```

Alternatively, as (rough) code:
```
package main

import (
        "context"
        "crypto/tls"
        "errors"
        "fmt"
        "net"

        "github.com/caddyserver/certmagic"
)

func main() {
        cfg := certmagic.NewDefault()

        if err := cfg.CacheUnmanagedCertificatePEMFile(context.TODO(), "/var/lib/acme/live/sub1.domain/fullchain", "/var/lib/acme/live/sub1.domain/privkey", []string{"cert0"}); err != nil {
                fmt.Println("cert0:", err)
        }
        if err := cfg.CacheUnmanagedCertificatePEMFile(context.TODO(), "/var/lib/acme/live/sub2.domain/fullchain", "/var/lib/acme/live/sub2.domain/privkey", []string{"cert1"}); err != nil {
                fmt.Println("cert1:", err)
        }

        result := func(t *tls.Certificate, err error) {
                if err != nil {
                        fmt.Println("failed:", err)
                } else {
                        fmt.Println("cert:", t.Leaf.Subject)
                }
        }

        fakeConn, _ := net.Dial("udp", "[::1]:1") // XXX: just so the log statement in GetCertificate doesn't crash

        cfg.CertSelection = certFromTag("cert0")
        result(cfg.GetCertificate(&tls.ClientHelloInfo{
                ServerName: "sub1.domain",
                Conn:       fakeConn,
        }))

        cfg.CertSelection = certFromTag("cert1")
        result(cfg.GetCertificate(&tls.ClientHelloInfo{
                ServerName: "sub2.domain",
                Conn:       fakeConn,
        }))
}

type certFromTag string

func (tag certFromTag) SelectCertificate(hello *tls.ClientHelloInfo, certs []certmagic.Certificate) (certmagic.Certificate, error) {
        for _, c := range certs {
                for _, t := range c.Tags {
                        if string(tag) == t {
                                return c, nil
                        }
                }
        }
        return certmagic.Certificate{}, errors.New("No matching certificate")
}
```
The above config and/or code example results in the certificate for the second domain not being found, without any error being returned by CacheUnmanagedCertificatePEMFile.

## What did you expect to happen, and what actually happened instead?

The certificate should be found; this scenario works fine in other webservers.

## How do you think this should be fixed?

Potentially, tags should be *added* to the existing certificate.
Or deduplication should not be just on certificate hash, but also account for tags.

**Comment by mholt** [comment]
That's a good point. I'll look into this!

**Comment by mholt** [comment]
I think this should be fixed now, in a46420b. Can you test and verify?
